File: analogagenda.client/src/app/DTOs/dev-kit.dto.ts
import { DevKitType, UsernameType } from "../enums";

export interface DevKitDto {
  id: string;
  name: string;
  url: string;
  type: DevKitType;
  purchasedBy: UsernameType;
  purchasedOn: string;
  mixedOn: string;
  validForWeeks: number;
  validForFilms: number;
  filmsDeveloped: number;
  imageUrl: string;
  description: string;
  expired: boolean
}

File: Database/DTOs/DevKitDto.cs
namespace Database.DTOs;

public class DevKitDto
{
    public string Id { get; set; } = string.Empty;

    public required string Name { get; set; }

    public required string Url { get; set; }

    public required string Type { get; set; }

    public required string PurchasedBy { get; set; }

    public DateOnly PurchasedOn { get; set; }

    public DateOnly MixedOn { get; set; }

    public int ValidForWeeks { get; set; }

    public int ValidForFilms { get; set; }

    public int FilmsDeveloped { get; set; }

    public string ImageUrl { get; set; } = string.Empty;

    public string Description { get; set; } = string.Empty;

    public bool Expired { get; set; }
}
